.pre-commit-config.yaml

```diff
@@ -0,0 +1,20 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-json
+      - id: check-merge-conflict
+  - repo: https://github.com/awslabs/git-secrets
+    rev: 80230afa8c8bdeac766a0fece36f95ffaa0be778
```

Comment:
Also, didn't catch this before, but no, we cannot pin to a specific commit like this. x_x

Reply:
Why? It's no less mutable than a tag. `git-secrets` has not been released since the pre-commit hook file was added therefore it's not possible to use a tag

Reply:
Because it's opaque. A tag tells you what to expect; a commit just tells you "idk, whatever this hash points to". How do I know if this is out of date and needs to be updated? How do I know that this is the right commit?

We should also not be relying on pre-release code except in exceptional cases. If there are changes to `git-secrets` that we want but haven't been released, we should work with the folks running that project to publish a new release.

Reply:
> How do I know if this is out of date and needs to be updated

The same way you would know with a tag - "is there one more recent than my current version?". Though it would mean passing the `--bleeding-edge` flag to [pre-commit autoupdate](https://pre-commit.com/#pre-commit-autoupdate)

> How do I know that this is the right commit?

The same way you'd know something is the right tag - you check what has changed since your current version before using it. The only commits since 1.3.0 have been doc formatting updates + the addition of pre-commit hook file.

> Because it's opaque. A tag tells you what to expect; a commit just tells you "idk, whatever this hash points to".

The same as "x.y.z" when there are no release notes or tag message. However, the tag is indicative of stability and an approach of only using known stable releases for deps makes sense (and makes all the above moot) so [I've requested a release](https://github.com/awslabs/git-secrets/pull/116) from `git-secrets`